rosidl_typesupport_opensplice_cpp/resource/msg__type_support.cpp.template

```diff
@@ -71,13 +73,17 @@ register_type__@(spec.base_type.type)(
   const char * type_name)
 {
   DDS::DomainParticipant * participant = static_cast<DDS::DomainParticipant *>(untyped_participant);
+  // TODO(wjwwood): use custom allocator; avoid new/delete.
   @(spec.base_type.pkg_name)::@(subfolder)::dds_::@(spec.base_type.type)_TypeSupport * ros_message_ts =
     new @(spec.base_type.pkg_name)::@(subfolder)::dds_::@(spec.base_type.type)_TypeSupport();
   DDS::ReturnCode_t status = ros_message_ts->register_type(participant, type_name);
   if (status != DDS::RETCODE_OK) {
-    printf("register_type() failed. Status = %d\n", status);
-    throw std::runtime_error("register_type failed");
+    std::stringstream ss;
+    ss << "register_type failed with status: " << status;
+    // TODO(wjwwood): use a custom exception
+    throw std::runtime_error(ss.str());
```

Comment:
That isn't sufficient since the way it is setup it does not pass any information back when it returns `false`. The whole function needs to be reimagined. I have TODO's to that effect in similar functions.